src/renderer/components/sidebar/view.jsx

```diff
@@ -1,56 +1,68 @@
 import React from 'react'
 import Icon from '@mdi/react'
 import classnames from 'classnames'
-import navigate from '@/utils/navigate'
-import Button from '@/components/common/button'
-import * as icons from '@/constants/icons'
+import {
+  HOME as home,
+  PLAYLIST_PLUS as playlistPlus,
+  HEART_OUTLINE as heart,
+} from '@/constants/icons'
 
-const NavLink = ({ icon, label, active, action, badge }) => {
-  return (
-    <div
-      onClick={action}
-      className={classnames('nav__link', { 'nav__link--active': active })}
-    >
-      <div className="link__body">
-        {icon && <Icon className="icon link__icon" path={icon} />}
-        {label && <span className="link__label">{label}</span>}
-      </div>
-      <div className="link__actions">
-        {badge && <div className="link__badge">{badge}</div>}
-      </div>
+const NavLink = ({ icon, label, active, action, badge }) => (
+  <div
+    onClick={action}
+    className={classnames('nav__link', { 'nav__link--active': active })}
+  >
+    <div className="link__body">
+      {icon && <Icon className="icon link__icon" path={icon} />}
+      {label && <span className="link__label">{label}</span>}
     </div>
-  )
-}
+    <div className="link__actions">
+      {badge && <div className="link__badge">{badge}</div>}
+    </div>
+  </div>
+)
 
 class SideBar extends React.PureComponent {
   constructor(props) {
     super(props)
   }
 
-  render() {
-    const { doNavigate, navigation, favorites } = this.props
+  getNavLink = ({ path, label, icon, idx }) => {
```

Comment:
Is this function really necessary?

Reply:
Did you test this? Other than that looks good :+1:

Reply:
I tested it locally. 

The getNavLinks reduces typing <NavLink> over and over which is better practice. https://en.wikipedia.org/wiki/Don%27t_repeat_yourself